integration/model_test.go

```diff
@@ -0,0 +1,156 @@
+// Copyright 2015 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package integration
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	cadvisor "github.com/google/cadvisor/info/v1"
+	"github.com/stretchr/testify/assert"
+
+	"k8s.io/heapster/manager"
+	model_api "k8s.io/heapster/model"
+	"k8s.io/heapster/sinks"
+	sink_api "k8s.io/heapster/sinks/api"
+	"k8s.io/heapster/sinks/cache"
+	source_api "k8s.io/heapster/sources/api"
+)
+
+type testSource struct {
+	createTimestamp time.Time
+}
+
+const (
+	podCount         = 10
+	testNamespace    = "testnamespace"
+	loadAverageMilli = 300
+)
+
+func (t testSource) buildPods(start time.Time) []source_api.Pod {
```

Comment:
Fuzzed structs are already used in unit tests to check corner cases. Here I would like to check if nothing is lost/dropped on the way if the values are correct. The values set below are (mostly) needed and the rest is irrelevant from the point of view of this test.